### PR TITLE
Feature/KITOSUDV-3321

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [ ] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [ ] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/Controllers/API/V1/OrganizationUnitLifeCycleController.cs [hunk 1]
```diff
@@ -32,5 +36,58 @@ public HttpResponseMessage Delete(Guid organizationUuid, Guid unitUuid)
                 .Delete(organizationUuid, unitUuid)
                 .Match(FromOperationError, Ok);
         }
+
+        [HttpGet]
+        [Route("{unitUuid}/access-rights")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage GetUnitAccessRights(Guid organizationUuid, Guid unitUuid)
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/OrganizationUnitLifeCycleController.cs:45]: not sure it belongs in a "lifecycle" controller... rights?

File: Presentation.Web/app/components/org/structure/org-structure.controller.ts [hunk 2]
```diff
@@ -632,17 +634,59 @@
...
+                $scope.isReordering = !$scope.isReordering;
+
+                if ($scope.isReordering) {
+                    $scope.loadingAccessRights = true;
+                    organizationUnitService.getUnitAccessRightsForOrganization(rootNodeOfOrganization.organization.uuid)
+                        .then(response => {
+                            applyAccessRights(rootNodeOfOrganization, response);
+                            $scope.loadingAccessRights = false;
+                        }, error => {
+                            notify.addErrorMessage("Failed to load unit access rights");
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-structure.controller.ts:650]: change text to danish

File: Presentation.Web/app/components/org/structure/org-structure.view.html [hunk 3]
```diff
@@ -36,25 +37,30 @@
...
+                        <div>
+                            <span class="org-structure-legend-color org-structure-legend-color-native-unit"></span> Enheder oprettet i KITOS
+                        </div>
+                        <div>
+                            <span class="org-structure-legend-color org-structure-legend-color-fk-org-unit"></span> Enheder synkroniseret fra FK Organisation
+                        </div>
                     </div>
                 </div>
+
+                <span ng-if="loadingAccessRights" class="pull-left"><progress-spinner data-show-text="true" data-custom-text="Loading unit access rights, it might take a while..." /></span>
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-structure.view.html:63]: change text to danish

File: Presentation.Web/app/models/api/organization/organization-unit-dto.ts [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+        name: string;
+        ean: string;
+        localId: number;
+        parentId: number;
+        organizationId: number;
+        origin: OrganizationUnitOrigin;
+        externalOriginUuid: string | null;
+        organization?: IOrganizationDto;
+        children?: IOrganizationUnitDto[];
+        draggable?: boolean;
```
Comment (a-naskret) [inline Presentation.Web/app/models/api/organization/organization-unit-dto.ts:13]: I had to add the properties to the Dto so I access them/assign the draggable. All of the nodes are of this type.

Or should I create a separate model, and map everything from the start?

File: Presentation.Web/Controllers/API/V1/OrganizationUnitController.cs [hunk 5]
```diff
@@ -121,8 +121,12 @@ public override HttpResponseMessage Patch(int id, int organizationId, JObject ob
                 {
                     //TODO: You have to be local or global admin to change parent
 
-                    var parentId = jtoken.Value<int>();
+                    if (!_orgUnitService.IsNativeKitosUnit(id))
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/OrganizationUnitController.cs:124]: Use the application service and GET the access rights for the unit. then you don't repeat the same logic here

File: Presentation.Web/app/components/org/structure/org-structure.controller.ts (hunk 2 above)
Comment (mrjsawdk) [reply Presentation.Web/app/components/org/structure/org-structure.controller.ts:650]: Kunne ikke indlæse rettighederne for organisationsenheden.

File: Presentation.Web/app/components/org/structure/org-structure.view.html (hunk 3 above)
Comment (mrjsawdk) [reply Presentation.Web/app/components/org/structure/org-structure.view.html:63]: Indlæser rettigheder (kan tage tid) ...

File: Presentation.Web/Controllers/API/V1/OrganizationUnitController.cs (hunk 5 above)
Comment (mrjsawdk) [reply Presentation.Web/Controllers/API/V1/OrganizationUnitController.cs:124]: it's only an error if the parent actually changed from the current version...
so in that case you probably need to load the entity from the bdatabase (not found if null) and THEN check if the incoming values are actually different from the current state before responding with an error

File: Tests.Unit.Core.ApplicationServices/ApplicationServices/Organizations/OrganizationUnitServiceTest.cs [hunk 6]
```diff
@@ -271,6 +271,22 @@ public void TransferSelectedOrganizationRegistrations_Returns_Forbidden(bool isU
             Assert.Equal(OperationFailure.Forbidden, result.Value.FailureType);
         }
 
+        [Fact]
+        public void GetAccessRightsByOrganization_Returns_NotFound_When_Organization_Was_NotFound()
+        {
+            //Arrange
+            var orgUuid = A<Guid>();
```
Comment (mrjsawdk) [inline Tests.Unit.Core.ApplicationServices/ApplicationServices/Organizations/OrganizationUnitServiceTest.cs:278]: what about the success case? I would assume you have already tested them using the GetAccess method tests, but we need at least one proof of it in this context as well. just testing "not found" is the poor mans choice

**Review by mrjsawdk — APPROVED** [review]
@a-naskret ready when tests are added

**Pull request merged**